### ISSUE TITLE
Validation sample_weights in Graph seems to be same as the target

### ISSUE THREAD

**Issue by bhokaal2k** [fchollet/keras#604]
May be I am missing something, but this piece of code in models.py L648-650 looks odd to me.

```python
if validation_data:
            sample_weight = [standardize_weights(validation_data[name]) for name in self.output_order]
            val_ins = [validation_data[name] for name in self.input_order] + [standardize_y(validation_data[name]) for name in self.output_order] + sample_weight
```
Why sample_weight and y are reading from the same field of validation data
```python
validation_data[name] for name in self.output_order
```

**Comment by fchollet** [comment]
They are not the same as the target, they're a all-ones array (i.e. no weights). 

There is no way at this time to use sample weights with `validation_data`, and it isn't clear what the API for that could look like (if you have any suggestion, let us know...). `validation_data` simply maps input/output names to data arrays. We could settle with a convention such as: "`validation_data['{name}_weights']` will contain the weights for output `name`", but that's pretty ugly.

**Comment by bhokaal2k** [comment]
Apologies for commenting on a closed topic, but could you please comment on the following for it's correctness. In the nutshell, I have two inputs (inp1 and inp2) and one output (label). Does the following call to fit function looks legit and is expected to do what I want i.e. train the model with class_weight defined by wt and validate using X_val_1, X_val_2 and Y_val using unweighted validation loss function. 
```python
model.fit({'inp1':X_1,'inp2':X_2,'label':Y_train},batch_size=2, nb_epoch=200, class_weight={'label':wt}, validation_data=({'inp1':X_val_1,'inp2':X_val_2,'label':Y_val}))
```
Thanks

**Comment by fchollet** [comment]
Yes, as far as I can tell this is correct.

**Comment by bhokaal2k** [comment]
Hi, 
   So, I just thought of a quick way of integrating validation weights to the current implementation, here is the code snipped based on what you already have in place and this one also has the same dependencies like building training_weights (indentations could be messed up)

```python
if validation_data or validation_split:
            val_f = self._test
        if validation_data:
            if 'class_weight' in validation_data:
                cw = validation_data['class_weight']
            else:
                cw = {}
            if 'sample_weight' in validation_data:
                sw = validation_data['sample_weight']
            else:
                sw = {}       
            sample_weight = [standardize_weights(validation_data[name],class_weight=cw[name],sample_weight=sw[name]) for name in self.output_order]
#             sample_weight = [standardize_weights(validation_data[name]) for name in self.output_order]
            val_ins = [validation_data[name] for name in self.input_order] + [standardize_y(validation_data[name]) for name in self.output_order] + sample_weight
```
And we make a call to fit is like before but with a dictionary inside the dictionary of validation_data 

```python
model.fit({'inp1':X_1,'inp2':X_2,'label':Y_train},batch_size=2, nb_epoch=200, class_weight={'label':wt}, validation_data=({'inp1':X_val_1,'inp2':X_val_2,'label':Y_val,'class_weight':{'label':None},'sample_weight':{'label':sampleWeight}}))

```
